### PR TITLE
Add allowOperators in DataClassContainsFunctions

### PR DESCRIPTION
**Pull request by atulgpt**
Fixes https://github.com/detekt/detekt/issues/5321

### CODE REVIEW

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/DataClassContainsFunctions.kt [hunk 1]
```diff
@@ -52,7 +56,7 @@ class DataClassContainsFunctions(config: Config = Config.empty) : Rule(config) {
         if (function.isOverride()) return
 
         val functionName = function.name
-        if (functionName != null && conversionFunctionPrefix.any { functionName.startsWith(it) }) return
+        if (functionName != null && (isAllowedConversionFunction(functionName) || checkOperator(function))) return
```
Comment (BraisGabin) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/DataClassContainsFunctions.kt:59]: I would move this check to another `if`. Checking that `functionName != null` and the operator in the same statement is a bit strange.

This way you can keep the old `if` as it was and just add a new one.

**Comment by BraisGabin** [comment]
Oh! And thanks for your contributions :)

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/DataClassContainsFunctionsSpec.kt [hunk 2]
```diff
@@ -43,6 +44,72 @@ class DataClassContainsFunctionsSpec {
...
+                assertThat(rule.compileAndLint(code)).isEmpty()
+            }
+
+            @Test
+            fun `does not report delegated operators if allowed`() {
+                val config = TestConfig(mapOf(ALLOW_OPERATORS to true))
+                val rule = DataClassContainsFunctions(config)
+                assertThat(rule.compileAndLint(code)).isEmpty()
+            }
+        }
```
Comment (BraisGabin) [inline detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/DataClassContainsFunctionsSpec.kt:110]: I'm not sure if I even like to have these tests here... This rule shouldn't care about delegation. And if it should we should handle that in a different issue/PR.

**Comment by atulgpt** [comment]
> Oh! And thanks for your contributions :)

Thanks :)

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/DataClassContainsFunctions.kt (hunk 1 above)
Comment (atulgpt) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/DataClassContainsFunctions.kt]: Made these changes

**Comment by atulgpt** [comment]
Hi @BraisGabin can you guide me why there is one check failing? Thanks

**Review by BraisGabin — APPROVED** [review]

**Pull request merged**